[OCPCLOUD-922] add alert for mhc in short circuit

elmiko opened the pull request:
This adds an alert for when an mhc has been in short-circuit for more
than 30 minutes, also adds some documentation about it.

ref: https://issues.redhat.com/browse/OCPCLOUD-922

michaelgugino reviewed (approved):
/lgtm

elmiko commented:
/cc @openshift/sre-alert-sme

ravitri commented on docs/user/Alerts.md:
```diff
@@ -69,3 +69,25 @@ due to either network issue or missing service definition.
+* The number of unhealthy machines has exceeded the `maxUnhealthy` limit for the check
+
+### Resolution
+Check to ensure that the `maxUnhealthy` field on the MachineHealthCheck is not set too low.
+In some cases a low value for `maxUnhealthy` will mean that the MachineHealthCheck will enter
+short-circuit if only a few nodes are unhealthy.
```
When we say, `not set too low`, I think the number would be different for small and large clusters. What about having a  recommendation for [maxUnhealthy](https://github.com/openshift/machine-api-operator/blob/master/pkg/apis/machine/v1beta1/machinehealthcheck_types.go#L63-L71) as a percentage value instead as an approximate and users can tune it further based on requirement?

dofinn replied:
+1 some gental guidance would be great here.

elmiko replied:
i'll add a little more here

dofinn commented on install/0000_90_machine-api-operator_04_alertrules.yaml:
```diff
@@ -52,3 +52,13 @@ spec:
+          labels:
+            severity: warning
+          annotation:
+            message: "machine health check {{ $labels.name }} has been short circuited for more than 30 minutes"
```
If im not familer with MHC and I see this alert for the first time, how do I know what short circuited is?

elmiko replied:
what if i change this to
```
machine health check {{ $labels.name }} has been disabled by short circuit for more than 30 minutes
```
would that make it clearer?
we do have a section in the official docs about short circuit, i'm just not sure how much to add in this alert message.

dofinn replied:
It does make it clearer. Thanks :) 
Good point on docs, we should not pack these `messages` with too much info. 
https://docs.openshift.com/container-platform/4.7/scalability_and_performance/recommended-cluster-scaling-practices.html#machine-health-checks-short-circuiting_cluster-scaling

dofinn commented:
/lgtm

elmiko commented:
updated
* change message in alert to reflect disabled status of mhc
* added more information to possible resolutions for this alert

cc @ravitri , let me know what you think of the new language in the alert doc

elmiko commented:
reminder: update https://github.com/openshift/enhancements/pull/538 once this merges

JoelSpeed commented:
/approve
/lgtm
/retest-required

JoelSpeed commented:
/override ci/prow/e2e-aws-operator

Currently failing on a known perma failure which we are working on a fix for, this alert doesn't appear to be causing issues so I believe we are safe for this to merge

JoelSpeed commented:
/override ci/prow/e2e-aws-operator

Based on comment above
